cpp/oneapi/dal/algo/logistic_regression/test/fixture.hpp

```diff
@@ -0,0 +1,219 @@
+            if (val < 0.5) {
+                *(y_ptr + i) = 0;
+            }
+            else {
+                *(y_ptr + i) = 1;
+            }
+        }
+    }
+
+    void run_test() {
+        std::int64_t train_size = n_ * 0.7;
+        std::int64_t test_size = n_ - train_size;
+
+        table X_train = homogen_table::wrap<float_t>(X_host_.get_mutable_data(), train_size, p_);
+        table X_test = homogen_table::wrap<float_t>(X_host_.get_mutable_data() + train_size * p_,
+                                                    test_size,
+                                                    p_);
+        table y_train =
+            homogen_table::wrap<std::int32_t>(y_host_.get_mutable_data(), train_size, 1);
+
+        const auto desc = this->get_descriptor();
+        const auto train_res = this->train(desc, X_train, y_train);
+        table intercept;
+        array<float_t> bias_host;
+        if (fit_intercept_) {
+            intercept = train_res.get_intercept();
+            bias_host = row_accessor<const float_t>(intercept).pull({ 0, -1 });
+        }
+        table coefs = train_res.get_coefficients();
+        auto coefs_host = row_accessor<const float_t>(coefs).pull({ 0, -1 });
+
+        std::int64_t train_acc = 0;
+        std::int64_t test_acc = 0;
+
+        const auto infer_res = this->infer(desc, X_test, train_res.get_model());
+
+        table resp_table = infer_res.get_responses();
+        auto resp_host = row_accessor<const std::int32_t>(resp_table).pull({ 0, -1 });
+
+        table prob_table = infer_res.get_probabilities();
+        auto prob_host = row_accessor<const float_t>(prob_table).pull({ 0, -1 });
+
+        for (std::int64_t i = 0; i < n_; ++i) {
+            float_t val = predict_proba(X_host_.get_mutable_data() + i * p_,
+                                        coefs_host.get_mutable_data(),
+                                        fit_intercept_ ? *bias_host.get_mutable_data() : 0);
+            std::int32_t resp = 0;
+            if (val >= 0.5) {
+                resp = 1;
+            }
+            if (resp == *(y_host_.get_mutable_data() + i)) {
+                if (i < train_size) {
+                    train_acc += 1;
+                }
+                else {
+                    test_acc += 1;
+                }
+            }
+            if (i >= train_size) {
+                REQUIRE(abs(val - *(prob_host.get_mutable_data() + i - train_size)) < 1e-5);
+                REQUIRE(*(resp_host.get_mutable_data() + i - train_size) == resp);
+            }
+        }
+        std::int64_t acc_algo = 0;
+        for (std::int64_t i = 0; i < test_size; ++i) {
+            if (*(resp_host.get_mutable_data() + i) ==
+                *(y_host_.get_mutable_data() + train_size + i)) {
+                acc_algo++;
+            }
+        }
+
+        float_t min_train_acc = 0.95;
+        float_t min_test_acc = n_ < 500 ? 0.7 : 0.85;
```

Comment:
What should it mean?

Reply:
For small datasets it is harder to fit model so accuracy threshold should be lower